tests/basetest.rb

```diff
@@ -28,60 +34,60 @@
   require File.expand_path('../../../cisco-nxapi/lib/cisco_nxapi')
 end
 
+# rubocop:disable Style/ClassVars
+# We *want* the address/username/password class variables to be shared
+# with all child classes, so that we only need to initialize them once.
+
 # TestCase - common base class for all minitest cases in this module.
 #   Most node utility tests should inherit from CiscoTestCase instead.
 class TestCase < Test::Unit::TestCase
-  # rubocop:disable Style/ClassVars
+  # These variables can be set in one of three ways:
+  # 1) ARGV:
+  #   $ ruby basetest.rb -- address username password
+  # 2) NODE environment variable
+  #   $ export NODE="address username password"
+  #   $ rake test
+  # 3) At run time:
+  #   $ rake test
+  #   Enter address or hostname of node under test:
   @@address = nil
   @@username = nil
   @@password = nil
-  # rubocop:enable Style/ClassVars
-
-  def process_arguments
-    if ARGV.length != 3 && ARGV.length != 4
-      puts 'Usage:'
-      puts '  ruby test_nxapi.rb [options] -- ' \
-           '<address> <username> <password> [debug]'
-      exit
-    end
-
-    # Record the version of Ruby we got invoked with.
-    puts "\nRuby Version - #{RUBY_VERSION}"
 
-    # rubocop:disable Style/ClassVars
-    @@address = ARGV[0]
-    @@username = ARGV[1]
-    @@password = ARGV[2]
-    # rubocop:enable Style/ClassVars
-
-    return unless ARGV.length == 4
-    if ARGV[3] == 'debug'
-      CiscoLogger.debug_enable
-    else
-      puts "Only 'debug' is allowed"
-      exit
-    end
-  end
-
-  # setup-once params
   def address
-    process_arguments unless @@address
+    @@address ||= ARGV[0]
+    @@address ||= ENV['NODE'].split(' ')[0] if ENV['NODE']
```

Comment:
It might make more sense to have ARGV parameters override the environment variable if parameters are present.

For example:

    set NODE="10.10.10.10 username password"
    ruby basetest.rb -- --address=11.11.11.11

Would use the username and password from the NODE variable, but it would replace the address with 11.11.11.11